src/network.ts

```diff
@@ -217,7 +217,12 @@ export async function launchNetwork() {
         network$.peerCount = peerCount;
     });
 
+    let lastValidatorRequest = { epoch: Infinity, ts: 0 };
```

Comment:
Maybe it's possible to store the current `fetch`'s `AbortController` in this object, too, so we can cancel pending requests when we get a new epoch.

Reply:
My suggestion is to type this as
```suggestion
    let currentValidatorRequest: { epoch: number, ts: Date, controller: AbortController } | undefined;
```

Having actual `Date` is much better for debugging.